graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -13,9 +13,74 @@
 #include <string>
 #include <vector>
 
+// Add for RandomEngine.
+#include <dmlc/thread_local.h>
+
+#include <pcg_random.hpp>
+#include <random>
+#include <thread>
+
 namespace graphbolt {
 namespace sampling {
 
+namespace {
+
+// Get a unique integer ID representing this thread.
+inline uint32_t GetThreadId() {
+  static int num_threads = 0;
+  static std::mutex mutex;
+  static thread_local int id = -1;
+
+  if (id == -1) {
+    std::lock_guard<std::mutex> guard(mutex);
+    id = num_threads;
+    num_threads++;
+  }
+  return id;
+}
+
+};  // namespace
+
+/**
+ * @brief Thread-local Random Number Generator class
+ */
+class RandomEngine {
```

Comment:
Is this Random Engine only designed for csc_sampling_graph or can be reused by other component in graphbolt?

Reply:
I believe this is a topic worth discussing because I think `RandomEngine` can be very helpful in any future scenarios where random numbers are required. Therefore, it would be beneficial to have it as a utility.

However, I am also worried about the **duplication** of this code, which can actually share the same code of `DGL`.